app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt

```diff
@@ -431,16 +427,27 @@ class NewMessageFragment : Fragment() {
         }
     }
 
-    private fun observeNewAttachments() = with(newMessageViewModel) {
-        importedAttachments.observe(viewLifecycleOwner) { (attachments, importationResult) ->
+    private fun observeAttachments() {
+        newMessageViewModel.attachmentsLiveData.observe(viewLifecycleOwner) {
+
+            val shouldTransition = attachmentAdapter.itemCount != 0 && it.isEmpty()
 
-            attachmentAdapter.addAll(attachments)
-            draftInRAM.attachments.addAll(attachments)
+            attachmentAdapter.submitList(it)
 
-            binding.attachmentsRecyclerView.isGone = draftInRAM.attachments.isEmpty()
+            binding.attachmentsRecyclerView.isVisible = if (it.isEmpty()) {
+                if (shouldTransition) TransitionManager.beginDelayedTransition(binding.root)
+                false
+            } else {
+                true
+            }
 
-            if (importationResult == ImportationResult.FILE_SIZE_TOO_BIG) showSnackbar(R.string.attachmentFileLimitReached)
-            updateIsSendingAllowed()
+            newMessageViewModel.updateIsSendingAllowed(attachments = it)
```

Comment:
This can be in the viewModel right? It's a bit weird to go back and forth between the fragment and viewModel.